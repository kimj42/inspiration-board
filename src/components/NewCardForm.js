import React, { Component } from 'react';
import PropTypes from 'prop-types';
import axios from 'axios';
import emoji from 'emoji-dictionary';
import './NewCardForm.css';

const EMOJI_LIST = ["", "heart_eyes", "beer", "clap", "sparkling_heart", "heart_eyes_cat", "dog"]

class NewCardForm extends Component {
  constructor() {
    super();

    this.state = {
      text: "",
      emoji: "",
    };
  }

  onFormSubmit = (event) => {
    console.log("I'm in onFormSubmit");
    // hold onto the this.state.text and this.state.emoji and add that to the card inside the board
    //then reset the state so that it's back to empty string
    event.preventDefault();
    const newCardData = {
      text: this.state.text,
      emoji: this.state.emoji,

    };

    this.setState({
      text: "",
      emoji: "",
    });

    return this.props.createNewCardCallback(newCardData);
  };

  onInputChange = (event) => {
    const field = event.target.name;
    const value = event.target.value;

    const newState = {};
    newState[field] = value;

    this.setState(newState);
  };

  // onInputChange = (event) => {
  //   console.log("In on input change");
  //
  //   const field = event.target.name;
  //   const value = event.target.value;
  //
  //   const newState = {};
  //   newState[field] = value;
  //   // newState.field = value;  // not the same :(
  //   this.setState(newState);
  // }

  render() {
    return (
      <section>
        <form onSubmit={this.onFormSubmit}>
          <div>
            <label htmlFor="text">Text: </label>
            <input name="text" type="text"
              value={this.state.text}
              onChange={this.onInputChange} />
          </div>
          <div>
            <label htmlFor="emoji">Emoji: </label>
            <input name="emoji" type="text"
              value={this.state.emoji}
              onChange={this.onInputChange} />
          </div>
          <input type="submit" value="Add Card" />
        </form>
      </section>
    );
  }
};

export default NewCardForm;
