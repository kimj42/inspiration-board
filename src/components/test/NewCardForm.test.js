import React from 'react';
import NewCardForm from '../NewCardForm.js';
import { shallow } from 'enzyme';

describe('NewCardForm', () => {
  test('see if newcardform works', () => {

  });
});
