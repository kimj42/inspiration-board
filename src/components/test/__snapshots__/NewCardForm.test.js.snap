// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewCardForm see if NewCardForm works without any props 1`] = `
<section>
  <form
    onSubmit={[Function]}
  >
    <div>
      <label
        htmlFor="text"
      >
        Text: 
      </label>
      <input
        name="text"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div>
      <label
        htmlFor="emoji"
      >
        Emoji: 
      </label>
      <input
        name="emoji"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <input
      type="submit"
      value="Add Card"
    />
  </form>
</section>
`;
